api/utils/aws/identifiers.go

```diff
@@ -61,6 +62,45 @@ func IsValidRegion(region string) error {
 	return trace.BadParameter("region %q is invalid", region)
 }
 
+const (
+	arnDelimiter    = ":"
+	arnPrefix       = "arn:"
+	arnSections     = 6
+	sectionService  = 2 // arn:<partition>:<service>:...
+	sectionAccount  = 4 // arn:<partition>:<service>:<region>:<accountid>:...
+	sectionResource = 5 // arn:<partition>:<service>:<region>:<accountid>:<resource>
+	iamServiceName  = "iam"
+)
+
+// CheckRoleARN returns whether a string is a valid IAM Role ARN.
+// Example role ARN: arn:aws:iam::123456789012:role/some-role-name
+func CheckRoleARN(arn string) error {
```

Comment:
nice. probably out of the scope of this PR but we probably can copy the entire arn.Parse here.

Reply:
The parse method returns the `arn.ARN` type, which would bring the AWS SDK to `api/go.mod` - I was trying to avoid that.
So, we would need to create a new type within the API (probably in this file).

It doesn't feel right to copy type/parser from the SDK into our own package :disappointed:

I was trying to copy the min required to ensure the error messages/logic stayed the same.

Reply:
> It doesn't feel right to copy type/parser from the SDK into our own package 😞

But we already copied like half of it =p. I guess we can review this another day. All arn-related stuff was supposed to be in `api` so we can do those checks in better places like `api.Database.CheckAndSetDefaults` but they ended up in lib because we don't want to bring AWS SDK to api/go.mod. I think this change is a good start 👍. Thanks!